sardana_albaem/ctrl/Albaem2OneDCtrl.py

```diff
@@ -0,0 +1,486 @@
+        self.itime = 0.0
+        self.master = None
+        self._latency_time = 0.001  # In fact, it is just 320us
+        self._repetitions = 0
+        self.formulas = {1: 'value', 2: 'value', 3: 'value', 4:'value'}
+
+        self._points_per_step = 1
+
+        self.lock = Lock()
+
+    @debug_it
+    def AddDevice(self, axis):
+        """Add device to controller."""
+        pass
+
+    @debug_it
+    def DeleteDevice(self, axis):
+        """Delete device from the controller."""
+        # self.albaem_socket.close()
+        pass
+
+    @debug_it
+    def PrepareOne(self, axis, value, repetitions, latency, nb_starts):
+        self._is_aborted = False
+
+    
+    @debug_it
+    def PreStateAll(self):
+        pass
+
+    @debug_it
+    def StateAll(self):
+        """Read state of all axis."""
+        state = self.sendCmd('ACQU:STAT?')
+
+        if state in ['STATE_ACQUIRING', 'STATE_RUNNING']:
+            self.state = State.Moving
+
+        elif state == 'STATE_ON':
+            self.state = State.On
+
+        elif state == 'STATE_FAULT':
+            self.state = State.Fault
+
+        else:
+            self.state = State.Fault
+        self.status = state
+
+    @debug_it
+    def StateOne(self, axis):
+        """Read state of one axis."""
+        return self.state, self.status
+
+    @debug_it
+    @handle_error(msg="LoadOne: Could not configure the device!")
+    def LoadOne(self, axis, value, repetitions, latency_time):
+        if axis != 1:
+            raise Exception('The master channel should be the axis 1')
+
+        self.itime = value
+
+        # Set Integration time in ms
+        val = self.itime * 1000
+        if val < 0.1:   # minimum integration time 
+            self._log.debug("The minimum integration time is 0.1 ms")
+            val = 0.1
+        self.sendCmd('ACQU:TIME %r' % val)
+
+        if self._synchronization in [AcqSynch.SoftwareTrigger,
+                                     AcqSynch.SoftwareGate]:
+            self._repetitions = 1
+            source = 'SOFTWARE'
+
+        elif self._synchronization == AcqSynch.HardwareTrigger:
+            source = 'HARDWARE'
+            self._repetitions = repetitions
+            if repetitions == 1:
+                self._repetitions = self._points_per_step
+        elif self._synchronization == AcqSynch.HardwareGate:
+            source = 'GATE'
+            self._repetitions = repetitions
+            if repetitions == 1:
+                self._repetitions = self._points_per_step
+        self.sendCmd('TRIG:MODE %s' % source)
+        if self._synchronization in [AcqSynch.HardwareTrigger,
+                                     AcqSynch.HardwareGate]:
+            self.sendCmd('TRIG:INPU %s' % self.ExtTriggerInput)
+        # Set Number of Triggers
+        self.sendCmd('ACQU:NTRI %r' % self._repetitions)
+        
+        # Array of arrays for ID readings from all channels
+        self.new_data = []
```

Comment:
just passing by this (you nostalgia ...) and i can not resist to point out that this line looks pointless :trollface:

Reply:
hahahhahhahaha

please, keep coming back!